### PR TITLE
feat(connector): [PowerTranz] Add template code for PowerTranz connector

### PR DESCRIPTION
**Pull request by ArjunKarthik**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Add template code for PowerTranz connector

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
For new connector PowerTranz integration

## How did you test it?
Template code testing not required

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/payments/flows.rs [hunk 1]
```diff
@@ -137,6 +137,7 @@ impl<const T: u8>
 }
 
 default_imp_for_complete_authorize!(
+    connector::Powertranz,
```
Comment (srujanchikke) [inline crates/router/src/core/payments/flows.rs:140]: this should be in alphabetical order .please change it in other places too

**Review by srujanchikke — CHANGES_REQUESTED** [review]

**Review by srujanchikke — APPROVED** [review]

File: crates/router/src/connector/powertranz/transformers.rs [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        }
+    }
+}
+
+//TODO: Fill the struct with respective fields
+// Auth Struct
+pub struct PowertranzAuthType {
+    pub(super) api_key: String,
```
Comment (SanchithHegde) [inline crates/router/src/connector/powertranz/transformers.rs:53]: Would prefer this being `Secret`.

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/router/src/connector/powertranz/transformers.rs (hunk 2 above)
Comment (ArjunKarthik) [reply crates/router/src/connector/powertranz/transformers.rs]: Done, I will also make changes in the template code as well in separate PR.

**Review by prasunna09 — APPROVED** [review]

**Pull request merged**